content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+Authorization Servers may support other profiles of OAuth if client does not include the `atproto` scope. For example, an Authorization Server might function as both an atproto PDS/entryway, and support other protocols/standards at the same time.
+
+Use of the atproto OAuth profile, as indicated by the `atproto` scope, means that the Authorization Server will return the atproto account DID as an account identifier in the `sub` field of token requests. A client may include only the `atproto` scope if they only need account authentication - for example a “Login with atproto” use case. Unlike OpenID, profile metadata in atproto is generally public, so an additional authorization scope for fetching profile metadata is not needed.
+
+In OAuth 2.0, Authorization Servers may always return the granted scopes in auth request responses and token responses, but are only required to do so if the scope is different from what the client requested. In the atproto OAuth profile, servers must always return the granted scopes.
+
+The intention is to support flexible scopes based on Lexicon namespaces (NSIDs) so that clients can be given access only to the specific content and API endpoints they need access to. Until the design of that scope system is ready, the atproto profile of OAuth defines two transitional scopes which align with the permissions granted under the original “session token” auth system:
+
+- `transition:generic`: broad PDS account permissions, equivalent to the previous “App Password” authorization level.
+    - write (create/update/delete) any repository record type
+    - upload blobs (media files)
+    - read and write any personal preferences
+    - API endpoints and service proxying for most Lexicon endpoints, to any service provider (identified by DID)
+    - ability to generate service auth tokens for the specific API endpoints the client has access to
+    - no account management actions: change handle, change email, delete or deactivate account, migrate account
+    - no access to DMs (the `chat.bsky.*` Lexicons), specifically
+- `transition:chat.bsky`: equivalent to adding the “DM Access” toggle for “App Passwords”
+    - API endpoints and service proxying for the `chat.bsky` Lexicons specifically
+    - ability to generate service auth tokens for the `chat.bsky` Lexicons
+    - this scope depends on and does not function without the  `transition:generic` scope
+
+### Authorization Requests
+
+This section details standards and requirements specific to Authorization Requests.
+
+PKCE and PAR are required for all client types and Authorization Servers. Confidential clients authenticate themselves using JWT client assertions.
+
+#### Request Fields
+
+A summary of fields relevant to authorization requests with the atproto OAuth profile:
```

Comment:
Specific case sensitivity for values here. Presumably they're all case sensitive, but best to call it out.

Reply:
Can we refer to / defer to an upstream OAuth RFC on this? there are probably many such details we aren't re-specifying in this document. would like to clarify if we are adding an atproto-specific requirement or not.